py4j-python/src/py4j/clientserver.py

```diff
@@ -203,6 +203,32 @@ class JavaClient(GatewayClient):
     ClientServerConnection per thread.
     """
 
+    class ThreadLocalConnectionCleaner(object):
+        """Cleans :class:`ClientServerConnection` in the thread local with
+        closing it properly, and removes it from the deque at
```

Comment:
Suggested change (but English is my second language too so take it with a grain of salt):

```
Cleans ... held by a thread local by closing it properly and removing it from the :class:`JavaClient` deque.

Right before the Python thread is terminated, this instance will be garbage-collected, which triggers a call to __del__  that contains the cleanup logic.
```